utils/gyb_syntax_support/TypeNodes.py

```diff
@@ -108,16 +125,22 @@
              Child('RightParen', kind='RightParenToken'),
          ]),
 
-    # tuple-type-element -> identifier? ':'? type-annotation ','?
-    Node('TupleTypeElement', kind='Syntax',
+    # throwing-specifier -> 'throws' | 'rethrows'
+    # function-type -> attribute-list '(' function-type-argument-list ')'
+    #   throwing-specifier? '->'? type?
+    Node('FunctionType', kind='Type',
          children=[
-             Child('Label', kind='IdentifierToken',
-                   is_optional=True),
-             Child('Colon', kind='ColonToken',
-                   is_optional=True),
-             Child('TypeAnnotation', kind='TypeAnnotation'),
-             Child('Comma', kind='CommaToken',
-                   is_optional=True),
+             Child('LeftParen', kind='LeftParenToken'),
+             Child('Arguments', kind='TupleTypeElementList'),
```

Comment:
Since I declared rich `TupleTypeElement`, I used `TupleTypeElementList` here instead of `FunctionTypeArgument List`.